src/graphics.c

```diff
@@ -47,6 +48,21 @@ static inline void plot_pixel(int x, int y, struct graphics_color c) {
     if (x < bounds_x_1 || x > bounds_x_2 || y < bounds_y_1 || y > bounds_y_2) {
         return;
     }
+
+    // Check to make sure that we are not drawing out of bounds
+    int diff_x = x - mouse_x;
+    int diff_y = y - mouse_y;
+    // if the point is in the mouse region, draw to the mouse buffer, copy into the video region, and draw the mouse
+    if (mouse_inited && diff_x < MOUSE_SIDE_2 && diff_x > -MOUSE_SIDE_2 && diff_y < MOUSE_SIDE_2 && diff_y > -MOUSE_SIDE_2) {
+        // wraps to [0, 28]
+        int x_index = diff_x + MOUSE_SIDE_2 - 1;
+        int y_index = diff_y + MOUSE_SIDE_2 - 1;
+        int index = y_index * (MOUSE_SIDE - 1) + x_index;
+        mouse_draw_buffer[index].r = c.r;
+        mouse_draw_buffer[index].g = c.g;
+        mouse_draw_buffer[index].b = c.b;
+        graphics_draw_mouse();
```

Comment:
If we are just managing the buffer behind the mouse here, do we need to draw the mouse again?